task_manager/task_manager.hh

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "replica/database.hh"
+#include "repair/row_level.hh"
+
+class task_manager {
+public:
+    enum class task_state {
+        RUNNING,
+        BLOCKED,
+        PAUSED,
+        DONE,
+        FAILED
+    };
+    static std::string serialize_state(task_state state) {
+        switch (state) {
+            case task_state::RUNNING: return "RUNNING";
+            case task_state::BLOCKED: return "BLOCKED";
+            case task_state::PAUSED: return "PAUSED";
+            case task_state::DONE: return "DONE";
+            case task_state::FAILED: return "FAILED";
+            default: return "";
+        }
+    }
+    struct task_status {
+        utils::UUID id;
+        std::string type;
+        task_state state;
+        bool is_abortable;
+        gc_clock::time_point start_time;
+        gc_clock::time_point end_time;
+        std::string error;
+        utils::UUID parent_id;
+        uint64_t number;
+        uint64_t shard;
+        std::string keyspace;
+        std::string table;
+        std::string entity;
+        std::string progress_units;
+        float progress_total;
+        float progress_completed;
```

Comment:
We need comments on these. Some fields here are self-explanatory, but many are not.